source/chapter2-devicetree-basics.rst

```diff
@@ -26,13 +26,20 @@ devices in a system. Each node has property/value pairs that describe
 the characteristics of the device being represented. Each node has
 exactly one parent except for the root node, which has no parent.
 
-A |spec|-compliant devicetree describes device information in a system
-that cannot necessarily be dynamically detected by a client program. For
-example, the architecture of PCI enables a client to probe and detect
-attached devices, and thus devicetree nodes describing PCI devices
-might not be required. However, a device node is required to describe a
-PCI host bridge device in the system if it cannot be detected by
-probing.
+A device in this context may be an actual hardware device, such as a UART. It
+may be part of a hardware device, such as the random-number generator in a TPM.
+It may also be a purely software concept, such as an bridge providing access to
```

Comment:
'purely software concept' is vague and leaves the door open to anything. 

I am assuming what you mean here is interfaces implemented firmware which provide access to h/w? I'd prefer something like 'A device may include functions implemented by firmware running in higher privilege levels or remote processors'.

Reply:
Yes that is a bit vague. See what you think of this version